src/extension.ts

```diff
@@ -3389,6 +3401,33 @@ async function startFlashing(
     flashType = await selectFlashMethod();
   }
 
+  Logger.info(`Using flash type: ${flashType}`, {tag: "Flash"});
+
+  try {
+    if(encryptPartitions) {
+      if(flashType !== ESP.FlashType.UART) {
+        throw new Error(`Invalid flash type for partition encryption. Required: UART, Found: ${flashType}`);
+      }
+      const eFuse = new ESPEFuseManager(workspaceRoot);
+      const summaryResult = await eFuse.readSummary();
+      if (summaryResult && summaryResult.BLOCK_KEY0) {
+        if (summaryResult.BLOCK_KEY0.value && summaryResult.BLOCK_KEY0.value == "00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00") {
```

Comment:
I would suggest to move the BLOCK_KEY0 default value to config.ts as a global variable and provide a meaningful name like empty eFuse or blank or something.